api/src/models/User.js

```diff
@@ -19,15 +19,25 @@ const userSchema = new Schema({
     email: {
         type: String,
         unique: true,
-        required: [true, 'Please add an email'],
+        required() {
+            if (this.registeredUsingOAuth) {
+                return false;
+            }
+            return [true, 'Please add an email'];
+        },
         match: [
-            /^[^\W][A-z0-9_]+(\.[a-zA-Z0-9_]+)*@[a-zA-Z0-9_]+(\.[a-zA-Z0-9_]+)*\.[a-zA-Z]{2,4}$/,
+            /(^[^\W][A-z0-9_]+(\.[a-zA-Z0-9_]+)*@[a-zA-Z0-9_]+(\.[a-zA-Z0-9_]+)*\.[a-zA-Z]{2,4}$)|/,
```

Comment:
I tried to do some smart work with this regex but, as you can see in the screenshot I link, it sucks.

![Screen Shot 2020-03-02 at 12 50 05 PM](https://user-images.githubusercontent.com/29370468/75673924-5cc3dd80-5c84-11ea-97e8-450f38b595ce.png)